src/main/java/ui/components/FilterTextField.java

```diff
@@ -70,15 +80,15 @@ private void setup(Predicate<String> validation) {
             if (key == null || key.isEmpty() || isModifierKeyPress) {
                 return;
             }
-            char typed = e.getCharacter().charAt(0);
-            // \b will allow us to detect deletion, but we can't find out the characters deleted
-            if (typed == '\t') {
-                e.consume();
-                if (!getSelectedText().isEmpty()) {
-                    confirmCompletion();
-                }
-            } else if (Character.isAlphabetic(typed) && shouldStartCompletion()) {
-                startCompletion(e);
+            
+            char typed = key.charAt(0);
+            
+            if (typed == '\t' && suggestion.isShowing()) {
+                suggestion.hide();
+                suggestion.getSelectedContent().ifPresent(this::completeWord);
+            } else {
+                suggestion.loadSuggestions(getMatchingKeywords(getCurrentWord()));
+                suggestion.show(this, Side.BOTTOM, 0, 0);
```

Comment:
Yup, so we can just have `suggestion.show(this)`. The details of where the menu is positioned are a concern of the menu.